pytorch_lightning/plugins/training_type/deepspeed.py

```diff
@@ -370,13 +370,6 @@ def init_deepspeed(self):
         precision = self.lightning_module.trainer.accelerator.precision
         model = LightningDeepSpeedModule(pl_module=self.model, precision=precision)
 
-        if self.zero_stage_3:
```

Comment:
Mind adding a test ? Not sure to understand the fix.

Reply:
hey @tchaton thanks for checking this out! 

Will do, I think this "fix" may break a case where not all parameters are defined in the `configure_sharded_model` hook + requires NVMe offloading. I'll make a test case to ensure we cover this :)